controllers/toolchaincluster/toolchaincluster_controller_test.go

```diff
@@ -151,9 +153,36 @@ func prepareReconcile(toolchainCluster *toolchainv1alpha1.ToolchainCluster, cl *
 		Client:     cl,
 		Scheme:     scheme.Scheme,
 		RequeAfter: requeAfter,
+		CheckHealth: func(context.Context, *kubeclientset.Clientset) []toolchainv1alpha1.Condition {
+			return toolchainCluster.Status.Conditions
+		},
```

Comment:
This is still hard to read to me. I still read it as "return the conditions the cluster already has". It feels a bit hacky. What if we pass this conditions in `prepareReconcile` instead? Something like the following:

```go
func prepareReconcile(toolchainCluster *toolchainv1alpha1.ToolchainCluster, cl *test.FakeClient, requeAfter time.Duration, healthConditions ...toolchainv1alpha1.Condition) (Reconciler, reconcile.Request) {
	controller := Reconciler{
		Client:     cl,
		Scheme:     scheme.Scheme,
		RequeAfter: requeAfter,
		CheckHealth: func(context.Context, *kubeclientset.Clientset) []toolchainv1alpha1.Condition {
			return healthConditions
		},
	}
	req := reconcile.Request{
		NamespacedName: test.NamespacedName(toolchainCluster.Namespace, toolchainCluster.Name),
	}
	return controller, req
}
```

WDYT?

Reply:
wouldnt it be the same..? 
we are going to pass conditions and i am passing the already available condition of the cluster..

Reply:
the main difference is that right now the cluster status is equal to the status returned by the healthcheck. After this change the actual status of the cluster can differ from the value returned by the healthcheck.

Reply:
basically we can not properly test the following scenarios:

```gherkin
Scenario: Conditions are updated
    Given a cluster with Condition C1
    When the healthcheck returns Condition C2
    Then the cluster is updated to have Condition C2
```

```gherkin
Scenario: Condition time is not updated
    Given a cluster with Condition C1
    When the healthcheck returns Condition C1
    Then the cluster has Condition C1
    And cluster Condition C1 LastTransitionTime is not updated
```

I think we should add these test case. I can not find them in the actual test suite.

Reply:
this will be a integration test , as this cannot be tested in unit tests, as heathcheck will be invoked from reconciler..?

Reply:
I have tried to add them